### ISSUE TITLE
MembershipReboot hybrid flow - Retrieve LinkedAccounts.ProviderAccountID

### ISSUE THREAD

**Issue by engenb** [IdentityServer/IdentityServer3#2606]
Hello,

#### To set the stage
I have requirements to authenticate web site users with Windows integrated auth and unfortunately, WS-Federation is not an option for me.  I have achieved this by creating an owin middleware that redirects authorize requests to a domain-joined, headless mvc action and configured this middleware as an external login provider in a standalone IdentityServer with MembershipReboot.  Everything is working great and I couldn't be happier.

When my middleware redirects to the domain-joined site, the user is given the 401 prompt.  If successful, I'm able to get the sid and username from the Request.LogonUserIdentity and return these values on the redirect back to IdSrv along with the state.  My middleware then uses the sid to query an api on the same domain-joined site to get more user information from ldap queries.  Basically, the sid is acting as my authorization code in a normal oauth flow.  As a side note, I'd be interested in any security ramifications you may see with this.  I realize auth codes expire and the sid will not, so that could be an issue.  I'm at a loss though for a better way to approach this without embedding yet another another IdSrv instance on that domain-joined windows login site to manage the tokens through client credential flows.

After fetching the user info I need, I create my external login ClaimsIdentity.  I'm providing the ldap objectId (a guid) as the NameIdentifier ClaimType.  This value is being stored in the IdSrv database as the LinkedAccounts.ProviderAccountID.

#### My question (finally):
I'd like to use the WindowsIdentity objectId (stored in IdSrv db as ProviderAccountID) to link the logged in user to other records in my system since this is a good unique identifier for the user.  I know I can simply create a custom claim for the user (I'm thinking external_sub) and store the objectId there as well, but I'd rather avoid that approach since this value is already stored elsewhere.  Is there a way to include this value in the UserInfo (or other) response?

The tricky part about it is, these other "records" I speak of will likely be created before the user ever logs in, and thus before the external user login exists in IdSrv - so I can't really use the sid or sub from the user as those are generated when the user is created.  I CAN however use that ldap objectId since that lives in the domain which can be queried (and trusted) when these "records" are created to set some kind of weak UserAccountId reference.

Thoughts on an approach?
Thanks for the great work on this awesome project!

**Comment by engenb** [comment]
I decided to add the ```external_provider_user_id``` ScopeClaim to the ```profile``` scope's claims.  Seeing as how this claim is in the IdSrv constants, is this the intended use of this claim?

**Comment by brockallen** [comment]
Well, the point of using IdSvr in front of your IdPs is to abstract them, but if your goal is not to abstract the windows identity then you can pass along the window's identity as any claim you want (perhaps even the subject claim). Not sure why you need MR in the middle either.

**Comment by engenb** [comment]
Exactly.  The goal here is to abstract the application from the underlying authentication mechanism.  The first "customer" of this product is internal to the company, so they want to use windows integrated auth with our domain.  Our next customer will likely have other authentication requirements so we thought it prudent to prepare for that.  Additionally, this website is half of the product.  The other half is a consumer-facing mobile application.  We wanted everyone authenticating through a common system.  IdSrv seemed a good fit for that.

We do want to abstract the logged in user.  The application shouldn't care if the user logged in with windows, google, or any other idp.  What I'm struggling with, is a good way to link these external users to records in my system.  I was hoping there was a general "best practice" around this.  Perhaps I use email?  Email is required so I can count on it being there, and it should be unique per user regardless of what idp is used to authenticate.

We went with MR over Identity Framework because it offers multi tenant features.  As I alluded above, the future of this project is to have multiple customers (tenants) all authenticating through IdSrv from their configured deployment of the web app.
